tests/unit/maxmemory.tcl

```diff
@@ -241,3 +241,29 @@ test_slave_buffers {slave buffer are counted correctly} 1000000 10 0 1
 # test again with fewer (and bigger) commands without pipeline, but with eviction
 test_slave_buffers "replica buffer don't induce eviction" 100000 100 1 0
 
+start_server {tags {"maxmemory"}} {
+    test {Can't rehash if used memory exceeds maxmemory after rehashing} {
+        r config set maxmemory 0
+        r config set maxmemory-policy allkeys-random
+        set buf ""
+        set rd [redis_deferring_client]
+
+        # Next rehash size is 8196, that will eat 64k memory
+        for {set i 0} {$i < 4096} {incr i} {
+            append buf "set $i $i\r\n"
+        }
+        $rd write $buf
```

Comment:
We do indeed wanna use a deferred client and a pipeline (so that population is quick), but there's no need for a single write here (probably taken from another test that needed a single event loop processing). 
Also you didn't actually read the responses, so you can't be sure the commands executed. 
In this case we prefer to use two simple loops, one using deferred client to execute commands (not write), and the second one to read the responses. Or maybe reuse a populate proc which we have somewhere (I rather stop using the debug populate command)

Reply:
Yes, you caught me, hhh
That you said alike this?
```
for {set i 0} {$i < 4096} {incr i} {
    $rd set $i $i
}
for {set i 0} {$i < 4096} {incr i} {
    $rd read ; # Discard replies
}
```

Reply:
yeah, but you can just use the `proc populate {num prefix size}` we now have in util.tcl.